### PR TITLE
Feat/tower-counter

### PR DESCRIPTION
**Pull request by Kegori228337**
created tower counter

### CODE REVIEW

File: Assets/Scripts/Tower/Wall.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Tower {
+	public class Wall : MonoBehaviour {
+		private Color _hoverColor;
+		private Renderer _render;
+		private Color _startColor;
+		private GameObject _tower;
+		public static int Counter;
+
+		private void Start() {
```
Comment (foxart) [inline Assets/Scripts/Tower/Wall.cs:11]: on start also update text for TowerCounter

File: Assets/Scripts/Tower/Tower.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using UnityEngine;
+
+namespace Tower {
+	public class Tower : MonoBehaviour {
+		[SerializeField] private GameObject bulletPrefab;
+		[SerializeField] private Transform firePoint;
+		private float _fireCountDown = 1f;
```
Comment (foxart) [inline Assets/Scripts/Tower/Tower.cs:7]: rename _fireRate

File: Assets/Scripts/Tower/Tower.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+using UnityEngine;
+
+namespace Tower {
+	public class Tower : MonoBehaviour {
+		[SerializeField] private GameObject bulletPrefab;
+		[SerializeField] private Transform firePoint;
```
Comment (foxart) [inline Assets/Scripts/Tower/Tower.cs:6]: remove serialized

File: Assets/Scripts/Tower/Tower.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+		private float _fireCountDown = 1f;
+
+		private void Update() {
```
Comment (foxart) [inline Assets/Scripts/Tower/Tower.cs:9]: don't use logic in built-in method,  move somewhere

Comment (foxart) [inline Assets/Scripts/Tower/Tower.cs:9]: investigate to use timeouts

File: Assets/Scripts/Tower/Bullet.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+namespace Tower {
+	public class Bullet : MonoBehaviour {
+		private const int Speed = 30;
+
+		private void Update() {
```
Comment (foxart) [inline Assets/Scripts/Tower/Bullet.cs:7]: remove logic from built-in method

File: Assets/Scripts/Tower/TowerCounter.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
+using UnityEngine;
```
Comment (foxart) [inline Assets/Scripts/Tower/TowerCounter.cs:1]: counter working when clicking on already occupied wall

File: Assets/Scripts/Tower/BuildManager.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
+using UnityEngine;
+
+namespace Tower {
+	public class BuildManager : MonoBehaviour {
+		public static BuildManager Instance;
+		[SerializeField] public GameObject towerPrefab;
+		private GameObject _towerToBuild;
```
Comment (foxart) [inline Assets/Scripts/Tower/BuildManager.cs:7]: remove unnecessary var

File: Assets/Scripts/Tower/BuildManager.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+		private void Awake() {
+			if (Instance != null) {
+				Debug.LogError("More than one BuildManager in scene");
+				return;
+			}
+
+			Instance = this;
+		}
+
+		private void Start() {
```
Comment (foxart) [inline Assets/Scripts/Tower/BuildManager.cs:18]: remove this logic

File: Assets/Scripts/Tower/BuildManager.cs [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+			_towerToBuild = towerPrefab;
+		}
+
+		public GameObject GetTowerToBuild() {
+			return _towerToBuild;
```
Comment (foxart) [inline Assets/Scripts/Tower/BuildManager.cs:23]: user towerPrefab

File: Assets/Scripts/Tower/BuildManager.cs [hunk 10]
```diff
@@ -0,0 +1,26 @@
+using UnityEngine;
+
+namespace Tower {
+	public class BuildManager : MonoBehaviour {
+		public static BuildManager Instance;
+		[SerializeField] public GameObject towerPrefab;
+		private GameObject _towerToBuild;
+
+		private void Awake() {
```
Comment (foxart) [inline Assets/Scripts/Tower/BuildManager.cs:9]: move this check logic to parent class Manager

File: Assets/Scripts/Tower/BuildManager.cs [hunk 11]
```diff
@@ -0,0 +1,26 @@
+using UnityEngine;
+
+namespace Tower {
+	public class BuildManager : MonoBehaviour {
```
Comment (foxart) [inline Assets/Scripts/Tower/BuildManager.cs:4]: rename to manager, also rename component in hierarchy

File: Assets/Scripts/Tower/Tower.cs (hunk 3 above)
Comment (Kegori228337) [reply Assets/Scripts/Tower/Tower.cs:6]: I can't I need this serialefields

**Review by foxart — APPROVED** [review]
job is done

**Review by dev4art — APPROVED** [review]

**Review by dev4art — APPROVED** [review]

**Pull request merged**